pkg/securitypolicy/framework.rego

```diff
@@ -0,0 +1,119 @@
+package framework
+
+import future.keywords.every
+import future.keywords.in
+
+default mount_device := false
+mount_device := true {
+    some container in data.policy.containers
+    some layer in container.layers
+    input.deviceHash == layer
+}
+
+layerPaths_ok(container) {
+    length := count(container.layers)
+    count(input.layerPaths) == length
+    every i, path in input.layerPaths {
+        container.layers[length - i - 1] == data.devices[path]
+    }
+}
+
+default mount_overlay := false
+mount_overlay := true {
+    some container in data.policy.containers
+    layerPaths_ok(container)
+}
+
+command_ok(container) {
+    count(input.argList) == count(container.command)
+    every i, arg in input.argList {
+        container.command[i] == arg
+    }
+}
+
+env_ok(pattern, "string", value) {
+    pattern == value
+}
+
+env_ok(pattern, "re2", value) {
+    regex.match(pattern, value)
+}
+
+envList_ok(container) {
+    every env in input.envList {
+        some rule in container.env_rules
+        env_ok(rule.pattern, rule.strategy, env)
+    }
+}
+
+workingDirectory_ok(container) {
+    input.workingDir == container.working_dir
+}
+
+default create_container := false
+create_container := true {
+    not input.containerID in data.started
+    some container in data.policy.containers
+    layerPaths_ok(container)
+    command_ok(container)
+    envList_ok(container)
+    workingDirectory_ok(container)
+    mountList_ok(container)
+}
+
+mountSource_ok(constraint, source) {
+    startswith(constraint, data.sandboxPrefix)
+    newConstraint := replace(constraint, data.sandboxPrefix, input.sandboxDir)
+    regex.match(newConstraint, source)
+}
+
+mountSource_ok(constraint, source) {
+    startswith(constraint, data.hugePagesPrefix)
+    newConstraint := replace(constraint, data.hugePagesPrefix, input.hugePagesDir)
+    regex.match(newConstraint, source)
+}
+
+mountSource_ok(constraint, source) {
+    constraint == source
+}
+
+mountConstraint_ok(constraint, mount) {
+    mount.type == constraint.type
+    mountSource_ok(constraint.source, mount.source)
```

Comment:
Requiring the mount source in the customer rego seems strange since it is forced to be a suffix wildcard with a few compiled in options for the prefix.  Any change in this implementation will break all policies, and there isn't any choice the customer can make.